XenAdmin/SettingsPanels/NRPEEditPage.cs

```diff
@@ -384,13 +376,20 @@ private void CheckDataGridView_BeginEdit(object sender, DataGridViewCellCancelEv
 
         private void CheckDataGridView_EndEdit(object sender, DataGridViewCellEventArgs e)
         {
-            DataGridViewCell currentCell = CheckDataGridView.CurrentRow.Cells[e.ColumnIndex];
-            if (!IsHost && currentCell.Value.ToString().Trim().Equals(""))
+            DataGridViewCell currentCell = CheckDataGridView.CurrentRow?.Cells[e.ColumnIndex];
+            
+            if (currentCell != null &&!_isHost && currentCell.Value.ToString().Trim().Equals(""))
             {
                 currentCell.Style.ForeColor = Color.FromKnownColor(KnownColor.ControlDark);
                 CheckGroupDictByLabel.TryGetValue(CheckDataGridView.CurrentRow.Cells[0].Value.ToString(), out CheckGroup checkGroup);
-                currentCell.Value = currentCell.ColumnIndex == 1 ? 
-                    checkGroup.WarningThresholdDefault : (object)checkGroup.CriticalThresholdDefault;
+
+                if (checkGroup != null)
+                {
+                    if (currentCell.ColumnIndex == WarningThresholdColumn.Index)

```

Comment:
When checking the column index it's best to check against WarningThresholdColumn.Index instead of the number, because if we add more columns in future or change the order of columns the code will break.